src/editors/containers/TextEditor/components/CodeEditor/index.jsx

```diff
@@ -26,17 +35,51 @@ export const hooks = {
       };
     }, []);
   },
+  cleanHTML: ({ initialText }) => {
+    const translateRegex = new RegExp(`&(${Object.keys(alphanumericMap).join('|')});`, 'g');
+    const translator = ($0, $1) => alphanumericMap[$1];
+    return initialText.replace(translateRegex, translator);
+  },
+  escapeHTMLSpecialChars: ({ ref, btnRef }) => {
+    const text = ref.current.state.doc.toString(); let
+      pos = 0;
+    const changes = [];
+    Object.keys(alphanumericMap).forEach(
+      (escapedKeyword) => {
+        // eslint-disable-next-line no-cond-assign
+        for (let next; (next = text.indexOf(alphanumericMap[escapedKeyword], pos)) > -1;) {
+          changes.push({ from: next, to: next + 1, insert: `&${escapedKeyword};` });
+          pos = next + 1;
+        }
+      },
+    );
+    ref.current.dispatch({ changes });
+    btnRef.current.setAttribute('disabled', 'disabled');
```

Comment:
Why are you disabling the button? Would we want the button to toggle instead? Also when it is disabled the button is no longer readable. I can see that it is still there but it is very difficult to read.

Reply:
Because this is an activity you can only do once. You don't want to exscape string literals witch, as that results in for example `&ldquo` -> `&ampldquo` (as it reads the ampersand), which isn't what we want.

Reply:
Do you want to remove the button from view instead of disabling it?

Reply:
Toggling is too much work for what our partners want (they either want one or the other.)

Reply:
Sure why not I'll do that?

Reply:
@KristinAoki I did that